byterun/caml/memory.h

```diff
@@ -79,6 +79,12 @@ color_t caml_allocation_color (void *hp);
 #define DEBUG_clear(result, wosize)
 #endif
 
+#ifdef COLLECT_STATS
+#define Count_alloc dom_st->allocations++
+#else
+#define Count_alloc while(0)
```

Comment:
I get what's going on here, but at first it looked like some magic was happening. `#define Count_alloc (void)0` or just `#define Count_alloc` are more idiomatic no-ops.